### ISSUE TITLE
SA 1202 confclicts with "Static member initializer refers to static member below or in other type part"

### ISSUE THREAD

**Issue by xxMUROxx** [StyleCop/StyleCop#51]
The rule SA 1202 **All private fields must be placed after all public fields** confclicts with **Static member initializer refers to static member below or in other type part**. This is clrearly a problem when writing readonly dependency properties:

Following code-snippet has _SA1202_

```
private static readonly DependencyPropertyKey MyPropertyKey = DependencyProperty.RegisterReadOnly(...);
public static readonly DependencyProperty MyProperty = MyPropertyKey.DependencyProperty;
```

Following code-snippet _Static member initializer refers to static member below or in other type part_

```
public static readonly DependencyProperty MyProperty = MyPropertyKey.DependencyProperty;
private static readonly DependencyPropertyKey MyPropertyKey = DependencyProperty.RegisterReadOnly(...);
```

The Problem is that stylecop's warning is just a warning. But if following this warning the code gets broken because the _DependencyProperty_ is then null;
